Draw Line

simonmcmullon opened the pull request:
Ive tried using drawline and can only get lines in one direction, that is going from top left to bottom right. If I try to get a line in the other direction, that is top right to bottom left the lines are ether horizontal or vertical.

Nkawu committed:
Initial commit

Nkawu committed:
Updated image URLs

Nkawu committed:
Changed README

Nkawu committed:
Renamed methods to line up with Adafruit libraries

Nkawu committed:
Added triangle draw methods

Nkawu committed:
First beta release

Nkawu committed:
Add 1 pixel column space between characters

Nkawu committed:
Update README.md

Nkawu committed:
Added dimensions image

techi602 committed:
Ardafruit GFX bitmap functions with example

Nkawu committed:
Merge pull request #1 from techi602/master

Ardafruit GFX bitmap functions with example

Nkawu committed:
Updated to use latest SPI library

Nkawu committed:
Merge pull request #2 from Nkawu/SW_SPI_Fix

Updated to use latest SPI library

Nkawu committed:
Create library.properties

Nkawu committed:
Update library.properties

Nkawu committed:
Update to Arduino IDE 1.5 Library specification

Nkawu committed:
Update library.properties

Nkawu committed:
Update library.properties

Nkawu committed:
Added Fast SPI bitbang code

Nkawu committed:
Update version & keywords

Nkawu committed:
Cleanup code

Nkawu committed:
Update README.md

ellsclytn committed:
Add PlatformIO library.json

Nkawu committed:
Merge pull request #3 from ellsclytn/master

Add PlatformIO library.json

Commit:
Fixed color table shift error

Nkawu committed:
Increment minor version

Commit:
Changes to avoid Windows IDE warnings

Commit:
Added support for variable backlight brightness

Commit:
Updated README

Commit:
Updated README with credit

Nkawu committed:
Merge pull request #6 from Nkawu/V1.2.3

V.1.2.3

Nkawu committed:
Update README.md

Nkawu committed:
Update library.json

Commit:
Open library to all platforms

Nkawu committed:
Update library.json

Nkawu committed:
Update library.properties

Nkawu committed:
Update README.md

Commit:
Reverted supported architectures to AVR only

Commit:
Increment version number

Nkawu committed:
Update README.md

Commit:
Adafruit_ILI9341 changes

Commit:
Added sample fonts

Commit:
Remove incorrect fonts

Commit:
Bug fixes

Commit:
Updated readme

Nkawu committed:
Merge branch 'master' into v1.2.6

Nkawu committed:
Merge pull request #10 from Nkawu/v1.2.6

V1.2.6

Commit:
Update README with install info

Nkawu committed:
Update README.md

Commit:
Fix ESP8266

Nkawu committed:
Merge pull request #11 from Nkawu/v1.2.7

v1.2.7

Nkawu committed:
Aesthetical define change

Nkawu committed:
V1.3.0 (#12)

* Initial checkin

* Add more fonts

* Updated examples

* Remove old files

* Renamed example

* Final changes

* M0 compatibility changes

* Removed unused variables

Nkawu committed:
Update README.md

Nkawu committed:
Update README.md

Nkawu committed:
Bug fixes for triangle functions (#14)

Nkawu committed:
V1.3.2 (#15)

* Add triangle demo

* Update README.md

* Update library.json

* Update library.properties

Nkawu committed:
Create CHANGES.md

Nkawu committed:
Update README.md

Nkawu committed:
Update README.md

Nkawu committed:
Update README.md

nicolasimeoni commented:
Hi,
I tried to use the library with STM32F1, using the stm32duino framework, but to allow it to compile I modified some file, described below. Can you update your library so it can compile with stm32duino?
regards
Niko

file: TFT_22_ILI9225.h:
line 12: #if defined(ARDUINO_STM32_FEATHER) || defined(ARDUINO_ARCH_STM32) || defined(ARDUINO_ARCH_STM32F1) || defined(__STM32F1__)

file: DefaultFonts.c:
line 7:  #if defined(ARDUINO_ARCH_SAM) || defined(ARDUINO_ARCH_SAMD) || defined(ARDUINO_ARCH_STM32) || defined(ARDUINO_ARCH_STM32F1)  || defined(__STM32F1__)

file Basic_demo:
#if defined (ARDUINO_ARCH_STM32F1)
  #define TFT_RST PA1 
  #define TFT_RS  PA2
  #define TFT_CS  PA0  // SS 
  #define TFT_SDI PA7  // MOSI
  #define TFT_CLK PA5  // SCK
  #define TFT_LED 0   // 0 if wired to +5V directly
#else
  #define TFT_RST 8 
  #define TFT_RS  9
  #define TFT_CS  10  // SS 
  #define TFT_SDI 11  // MOSI
  #define TFT_CLK 13  // SCK
  #define TFT_LED 3   // 0 if wired to +5V directly
#endif

Nkawu committed:
Added support for STM32F1 (#18)

Nkawu committed:
Add support for ESP32 (#19)

Commit:
Stm32 f1 support and speeding up the library. (#20)

* initial Commit stm32f1

* 1.Variante mit Char-Window

Buchstaben werden in einem Window mit autoinrement-decrement gezeichnet,
ohne direkte x-Y Adressierung

* selection of string class possible

by means of a define in the .h file strings kann be char Arrays or
Strings

* errorcorrection for AVR

Wrong driving SPI

* preparing faster bitmöas

* bimapOptimze-1

use autoincrement when drawing bitmals

* bitmap drawing accelerated

in all 4 variants. All 4 varaints are combined in one function

* reset defaultFonts

* XBitmap optimized

Drawing of Xbitmaps integrated in std-Bitmaps drawings.

* monospaced-Flag added

when setting a font, the monospace flag forces equidistant character
spacing.

* process clipped characters correctly

clipped charachters are drawn with drawPixel commands

* updated keywords.txt

* getFont Function

* update defaultFonts to master-version

* SW-SPI reenabled

SW SPI is working again now

Nkawu committed:
v1.3.5 (#24)

Nkawu committed:
V1.3.6 (#25)

* High speed color bitmap function

* Corrected param

* fixed _setWindow call

* Add high speed color drawBitmap function

Nkawu committed:
v1.3.7

Fix esp8266 type error with min() (#27)

Nkawu committed:
Fix MD typo

per1234 committed:
Use correct field separator in keywords.txt (#29)

The Arduino IDE requires the use of a single true tab separator between the keyword name and identifier. When spaces are used rather than a true tab the keyword is not highlighted.

Reference:
https://github.com/arduino/Arduino/wiki/Arduino-IDE-1.5:-Library-specification#keywords

Bolukan committed:
State corrupted after a direct call to drawLine (#33)

Solved the corrupt state of checkSPI after a direct drawLine call.

Nkawu committed:
Create version 1.3.8

Bolukan committed:
Add image of ILI9225 variant for use in Wiki (#37)

Bolukan committed:
Add advised pins for ESP8266 and ESP32 (#40)

Bolukan committed:
Add advised pins for ESP8266 and ESP32 (#38)

Bolukan committed:
Add advised pins for ESP8266 and ESP32 (#39)

Nkawu committed:
v1.3.9 (#41)

Bolukan committed:
Added HSPI support for ESP32 (#42)

G4lile0 committed:
fixed typo for linux Math.h --> math.h (#31)

Bolukan committed:
Replaced checkSPI with writeFunctionLevel to make use of startWrite/endWrite more robust (#44)

Nkawu committed:
Add pull requests (#47)

joe820730 committed:
Add 1-D bitmap array support and fix drawing issue. (#48)

* Add 1-D bitmap array support.

* Fix drawing issue due to wrong _setWindow() args.

Nkawu committed:
v1.4.1 (#49)

LAK132 committed:
More performance improvements for color drawBitmap (#53)

* Add _spiWrite16

* Use _spiWrite16 in the drawBitmap functions

* Use HSPI_WRITE_PIXELS in 1D color drawBitmap

* Add _spiWrite16

* 1D drawBitmap should have own description

* j iteratoes over h, so it should be j*w not j*h

* Only actually need one loop

* Add HSPI_WRITE_PIXELS to 2D color drawBitmap

* Fix indentation

Nkawu committed:
v1.4.2 More performance improvements for color drawBitmap (#54)

Nkawu committed:
Update mds

Nkawu committed:
Update mds

Nkawu committed:
Fix MDs and code layout (#55)

Nkawu committed:
Fix MDs

Nkawu committed:
Fix MDs

Nkawu committed:
Fix MDs

Nkawu committed:
Fix MDs

Nkawu committed:
Fix MDs

Nkawu committed:
Change image

Nkawu committed:
Change image

Nkawu committed:
Change image

Nkawu committed:
Fix typo add Wiki link

Nkawu committed:
More doc text

Nkawu committed:
Create LICENSE

dl9rdz committed:
Using FAST_PINIO on dual-core ESP32 is causing fatal errors due to side effects. The (#62)

operations on the port are non-atomic, and can have unwanted and unexpected side effects
on software running on the other CPU core if this software also accesses IO ports.

In addition, FAST_PINIO is also not fast on ESP32. My reference board (T-Beam v1.0)
takes 2.48s to flip a pin 10e7 times with digitalWrite(pin, ON); digitalWrite(pin, OFF).
It takes 2.52s to flip the same pin 10e7 times with *pin |= pinmask; *pin &= ~pinmask.

So this modification causes a slight speed-up and removes serios problems.

Bolukan committed:
replace SPI_WRITE16 (#63)

Cannot compile with unknown SPI_WRITE16

Bolukan committed:
remove compiler warnings (#64)

* remove compiler warnings

* Remove unused variable
Use HSPI_WRITE

Nkawu committed:
v1.4.3

charclo committed:
arduino nano every support

Nkawu committed:
Merge branch 'charclo-nano-every'

Nkawu committed:
v1.4.4

pierbout committed:
Type uint32_t error (#71)

I think there is a small type error on line 28 for STM32 boards: uint32 --> uint32_t

Nkawu committed:
Update examples for ESP32
